UnityProject/Assets/Scripts/Objects/VisibleBehaviour.cs

```diff
@@ -93,7 +95,20 @@ private void UpdateState(bool _aliveState)
 
 		for (int i = 0; i < renderers.Length; i++)
 		{
-			renderers[i].enabled = _aliveState;
+			SpriteRenderer sr = renderers[i] as SpriteRenderer;
+			// Cast and check cast.
+			// This is necessary because some renderers fail the cast.
+			if (sr != null)
+			{
+				if (CanBeDisabled(sr))
+				{
+					renderers[i].enabled = _aliveState;
+				}
+			}
+			else
+			{
+				renderers[i].enabled = _aliveState;

```

Comment:
Why do the if and the else have the same result?

Reply:
I needed to get just the sprite renderers to test them for eligibility. The other renderers need no filtering.

Reply:
if(sr != null && !CanBeDisabled(sr))
{
*Do nothing*
}

Else
{
renderers[i].enabled = _aliveState;
}

Reply:
Oh ok I see

Reply:
Good advise:
If you would tell it to a person as: "I want X only to happen when Y"
That would be your IF and the rest will be put into ELSE.